File: bowling_exercise/python/game_test.py
import unittest
from game import Game


class TestGame(unittest.TestCase):
    def test_gutter_ball_game(self):
        game = Game()

        # 10 frames, 2 rolls each = 20 gutter balls.
        for _ in range(20):
            game.roll(0)

        # End of the game!
        self.assertEqual(0, game.score())

if __name__ == '__main__':
    unittest.main()

File: bowling_exercise/ruby/game_test.rb
$:.unshift File.dirname(__FILE__)

require "minitest/autorun"
require "game"

class TestGame < MiniTest::Unit::TestCase
  def test_gutter_ball_game
    game = Game.new

    # 10 frames, 2 rolls each = 20 gutter balls.
    20.times do
      game.roll(0)
    end

    # End of the game!
    assert_equal(0, game.score())
  end
end